addons/uom/models/uom_uom.py

```diff
@@ -24,12 +24,7 @@ def _onchange_uom_ids(self):
         else:
             reference_count = sum(uom.uom_type == 'reference' for uom in self.uom_ids)
             if reference_count == 0 and self._origin.id:
-                return {
-                    'warning': {
-                        'title': _('Warning!'),
-                        'message': _("UoM category %s should have a reference unit of measure.") % self.name
-                    }
-                }
+                raise UserError(_("UoM category %s must have atleast one reference unit of measure.") % self.name)
```

Comment:
raise UserError(_('UoM category %s must have atleast one reference unit of measure.', self.name))